### DISCUSSION TITLE
Launch Actions (Fish & macOS)

### DISCUSSION THREAD

**Discussion by karimlevallois** [kovidgoyal/kitty#7132] [Q&A] [answered]
Hi everyone!

So I've moved to fish on macOS and set it as my default shell etc. Kitty sees it as the default shell and everything is working as expected, except one thing.

I'm using launch-actions.conf to open files within nvim in Kitty, but for the life of me I cannot get it to work with fish. It does however open if I add 'zsh' inside the action (see below), but obviously I shouldn't have to fall-back to zsh.

```
# Open text files without fragments in the editor
protocol file
mime text/*
action launch --type=os-window zsh $EDITOR $FILE_PATH
```

In my startup.conf I know I have to use `launch fish -c 'nvim'` to get it to work and have tried many variations of this inside the launch-actions.conf too, but to no avail.

Any help greatly received.

Many thanks, Karim

**Comment by kovidgoyal** [comment]
You dont need to use any shell inside a launch action just run your
editor directly. And the zsh invocation you showed wont work either, you
need -c there as well. Finally, generally speaking use absolute paths
when running things.

**Reply by karimlevallois** [reply]
Thanks for taking the time to reply. 

You are correct that nvim will open without adding in the shell with `action launch --type=os-window $EDITOR $FILE_PATH`, the issue I’m having is that it doesn’t open the actual file, it just shows the nvim dashboard screen as if I’ve just called `nvim` alone. So the `$FILE_PATH` part doesn’t seem to work. The only time I can get the actual file to open in nvim is to add `zsh` in.

**Comment by kovidgoyal** [comment]
FILE_PATH is expanded by *kitty* *before* invoking the command. Adding
zsh to it cannot possibly make any difference. And 
zsh xyz abc
will anyway try to open xyz as a shell script in zsh. It wont run xyz.
I dont know what you are actually doing but it cant be what you say you are doing.

**Reply by karimlevallois** [reply]
It doesn’t make sense I agree, but it only opens the file when I add it. Happy to take some videos showing the config and what’s happening.

Is there any reason why FILE_PATH wouldn’t work in opening the file?

**Comment by kovidgoyal** [comment]
On Thu, Feb 15, 2024 at 09:48:54PM -0800, Karim Le Vallois wrote:
> It doesn’t make sense I agree, but it only opens the file when I add it. Happy to take some videos showing the config and what’s happening.
> 
> Is there any reason why FILE_PATH wouldn’t work in opening the file?

Only one I can think of is if it is not an absolute path and the CWD is
being changed when nvim starts.

**Reply by karimlevallois** [reply]
OK, so must be fish related then as I wasn’t getting this issue before switching to it. This is why I tried `zsh` by the way, to see if it was shell related rather than something that had changed in Kitty.

**Comment by karimlevallois** [comment]
OK, seem to just be LunarVim that I have issues with. Launches work fine with Neovim and Helix, so must be something funky with LunarVim.

**Comment by karimlevallois** [comment, marked as answer]
If anyone else comes across this issue themselves, I fixed it with the following...

```
# Open text files without fragments in the editor.
protocol file
mime text/*
action launch --type=tab fish -c '$EDITOR $FILE_PATH'
```
